sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c

```diff
@@ -0,0 +1,263 @@
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 2 (i.e. reset).
+ * Also program the aon timer with:
+ * - bark after escalation starts, so the interrupt is suppressed by escalation,
+ * - bite after escalation reset, so we should not get timer reset.
+ */
+enum {
+  kWdogBarkMicros = 3 * 100,          // 300 us
+  kWdogBiteMicros = 8 * 100,          // 800 us
+  kEscalationPhase0Micros = 2 * 100,  // 200 us
+  // The cpu value is set slightly larger, since if they are the same it is
+  // possible busy_spin_micros in execute_test can complete before the
+  // interrupt.
+  kEscalationPhase0MicrosCpu = kEscalationPhase0Micros + 20,  // 220 us
+  kEscalationPhase1Micros = 2 * 100,                          // 200 us
+  kEscalationPhase2Micros = 100,                              // 100 us
+};
+
+static_assert(kWdogBarkMicros < kWdogBiteMicros &&
+                  kWdogBarkMicros > kEscalationPhase0Micros &&
+                  kWdogBarkMicros <
+                      (kEscalationPhase0Micros + kEscalationPhase1Micros) &&
+                  kWdogBiteMicros >
+                      (kEscalationPhase0Micros + kEscalationPhase1Micros),
+              "The wdog bite shall happen only if escalation reset fails.");
+
+/**
+ * Objects to access the peripherals used in this test via dif API.
+ */
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+static dif_clkmgr_t clkmgr;
+static dif_rstmgr_t rstmgr;
+static dif_alert_handler_t alert_handler;
+
+volatile bool interrupt_seen = false;
+
+/**
+ * External ISR.
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this OTTF ISR. This ISR
+ * overrides the default OTTF implementation.
+ */
+void ottf_external_isr(void) {
+  interrupt_seen = true;
+  dif_rv_plic_irq_id_t irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    LOG_ERROR("Unexpected aon timer interrupt %0d", irq);
```

Comment:
nit: Should this be a CHECK evaluating to false to abort the test case?

Reply:
Done.